server/app/tasks/DatabaseSeedTask.java

```diff
@@ -0,0 +1,162 @@
+                      Lang.forCode("es-US").toLocale(),
+                          "Nombre (nombre y la inicial del apellido está bien)",
+                      Lang.forCode("zh-TW").toLocale(), "姓名（名字和姓氏第一個字母便可）")),
+              /* questionHelpText= */ LocalizedStrings.empty()),
+          new DateQuestionDefinition(
+              /* name= */ "Applicant Date of Birth",
+              /* enumeratorId= */ Optional.empty(),
+              /* description= */ "Applicant's date of birth",
+              /* questionText= */ LocalizedStrings.of(
+                  Lang.forCode("en-US").toLocale(),
+                  "Please enter your date of birth in the format mm/dd/yyyy"),
+              /* questionHelpText= */ LocalizedStrings.empty()));
+
+  private final Provider<QuestionService> questionServiceProvider;
+  private final Provider<VersionRepository> versionRepository;
+  private final Database database;
+
+  @Inject
+  public DatabaseSeedTask(
+      Provider<QuestionService> questionService, Provider<VersionRepository> versionRepository) {
+    this.questionServiceProvider = checkNotNull(questionService);
+    this.versionRepository = checkNotNull(versionRepository);
+    this.database = DB.getDefault();
+
+    this.run();
+  }
+
+  private void run() {
+    if (isMissingCanonicalQuestions()) {
+      // Ensure a draft version exists to avoid transaction collisions.
+      versionRepository.get().getDraftVersion();
+      inSerializableTransaction(this::seedCanonicalQuestions, 1);
+    }
+  }
+
+  /**
+   * Ensures that questions with names matching those in {@code CANONICAL_QUESTIONS} are present in
+   * the database, inserting the definitions in {@code CANONICAL_QUESTIONS} if any aren't found.
+   */
+  private void seedCanonicalQuestions() {
+    var questionService = questionServiceProvider.get();
+    ImmutableSet<String> existingQuestionNames = questionService.getQuestionNames();
+
+    for (QuestionDefinition questionDefinition : CANONICAL_QUESTIONS) {
+      if (existingQuestionNames.contains(questionDefinition.getName())) {
+        LOGGER.info(
+            "Canonical question \"%s\" exists at server start", questionDefinition.getName());
+        continue;
+      }
+
+      ErrorAnd<QuestionDefinition, CiviFormError> result =
+          questionService.create(questionDefinition);
+
+      if (result.isError()) {
+        String errorMessages =
+            result.getErrors().stream()
+                .map(CiviFormError::message)
+                .collect(Collectors.joining(", "));
+
+        LOGGER.error(
+            String.format(
+                "Unable to create canonical question \"%s\" due to %s",
+                questionDefinition.getName(), errorMessages));
+      } else {
+        LOGGER.info("Created canonical question \"%s\"", questionDefinition.getName());
+      }
+    }
+  }
+
+  private void inSerializableTransaction(Runnable fn, int tryCount) {
```

Comment:
Note to reviewers:

I can imagine this maybe (??) causing livelock if multiple servers start at near enough the same time and try to seed at near enough the same time, I think limiting the number of retries should mean that in such a circumstance eventually all of the servers except one will run out of retries in which case the last server standing wins and gets to commit its transaction.

Other things we could do here to reduce this problem:

A. Use the [akka scheduler](https://www.playframework.com/documentation/2.8.x/ScheduledTasks) to reschedule retries instead of recursing. Could reschedule with some random jitter to try and reduce the likelihood of servers scheduling their retries at the same time.

B. Schedule the initial attempt at a random delay. The downside there is that the first request to a rebuilt dev server won't trigger seeding so if you clear your questions in dev and reload the questions index it won't have the canonical questions until you reload again.

A and B aren't mutually exclusive.

Reply:
Just to confirm, this task wouldn't block the server being considered healthy, correct? If we were to block serving requests on new server startup / restart, it'd be unfortunate for a livelock situation to cause perceived slowness in responding to requests.

Reply:
It will not.